modules/custom/activity_creator/src/ActivityNotifications.php

```diff
@@ -125,36 +151,92 @@ public function markEntityNotificationsAsRead(AccountInterface $account, EntityB
    * @param \Drupal\Core\Entity\EntityBase $entity
    *   Entity object.
    *
-   * @deprecated in opensocial:8.x-7.0 and is removed from opensocial:8.x-7.1. Use
+   * @deprecated in opensocial:8.x-7.1 and is removed from opensocial:8.x-8.0. Use
    *   \Drupal\activity_creator\ActivityNotifications
    * ::markEntityNotificationsAsRead() instead.
    *
    * TODO: Change @see to point to a change record.
    * @see https://www.drupal.org/project/social/issues/3087083
    */
   public function markEntityAsRead(AccountInterface $account, EntityBase $entity) {
-
     // Retrieve all the activities referring this entity for this account.
     $ids = $this->getNotificationIds($account, [ACTIVITY_STATUS_RECEIVED, ACTIVITY_STATUS_SEEN], $entity);
-    $this->changeStatusOfActivity($ids, ACTIVITY_STATUS_READ);
+    $this->changeStatusOfActivity($ids, $account, ACTIVITY_STATUS_READ);
 
   }
 
+  /**
+   * Mark an activity notification as read for a given account.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   Account object.
+   * @param \Drupal\activity_creator\Entity\Activity $activity
+   *   Activity object.
+   *
+   * @return bool
+   *   TRUE or FALSE depending upon update status.
+   */
+  public function markNotificationAsRead(AccountInterface $account, Activity $activity): bool {
+    // Change the activity notification status to read.
+    return $this->changeStatusOfActivity([$activity->id()], $account, ACTIVITY_STATUS_READ);
+
+  }
+
+  /**
+   * Mark all acitivty notification as read for a given account.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   Account object.
+   *
+   * @return bool
+   *   TRUE or FALSE depending upon update status.
+   */
+  public function markAllNotificationAsRead(AccountInterface $account) {
+    // Retrieve all the activities referring this entity for this account.
+    if (!empty($ids = $this->getNotificationIds($account, [ACTIVITY_STATUS_RECEIVED, ACTIVITY_STATUS_SEEN]))) {
+      return $this->changeStatusOfActivity($ids, $account, ACTIVITY_STATUS_READ);
+    }
+
+    $this->loggerFactory->get('activity_creator')->notice('There are no notification to be marked as read for user @id', [
```

Comment:
@bramtenhove I am wondering that logging after a negative response from a service's method is good idea or it should be done by party whose calling the service w.r.t to the response from service?